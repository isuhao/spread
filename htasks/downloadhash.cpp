#include "downloadhash.hpp"
#include "tasks/download.hpp"

using namespace Spread;

Jobs::Job* DownloadHash::createJob()
{
  // Guess the hash from targets
  assert(outputs.size() > 0);
  const Hash &hash = outputs.begin()->first;
  return new Tasks::DownloadTask(url, getOutStream(hash));
}
