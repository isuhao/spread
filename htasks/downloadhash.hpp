#ifndef __HASH_DOWNLOADTASK_HPP_
#define __HASH_DOWNLOADTASK_HPP_

#include "hashtask.hpp"

/* This class downloads one file to a given output set. The hash is
   not set directly, instead it is guessed from the first entry in the
   download list.
 */

namespace Spread
{
  struct DownloadHash : HashTask
  {
    DownloadHash(const std::string &_url)
      : url(_url) {}

  private:
    std::string url;
    Jobs::Job *createJob();
  };
};

#endif
