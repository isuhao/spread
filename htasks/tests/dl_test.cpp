#include "downloadhash.hpp"

#include <iostream>
using namespace std;
using namespace Spread;
using namespace Jobs;

int main()
{
  string url = "http://tiggit.net/robots.txt";
  cout << "Downloading " << url << " ...\n";

  Hash hash("N0VT8AYfLEu2hFufocgj9ykAQoNEgcQwzLW7m1Tfc-cj");

  DownloadHash dl;
  dl.url = url;
  dl.addOutput(hash, "_robots.txt");
  dl.addOutput(hash, "_robots2.txt");
  JobInfoPtr info = dl.run(false);
  cout << info->getCurrent() << "/" << info->getTotal() << endl;

  return 0;
}
