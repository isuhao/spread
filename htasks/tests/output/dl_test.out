Downloading http://tiggit.net/robots.txt ...
35/35
